nemo/collections/asr/parts/preprocessing/perturb.py

```diff
@@ -999,7 +1185,12 @@ class CustomPerturbation(perturb.Perturbation):
                 raise ValueError("`prob` must be a float value between 0 and 1.")
 
             try:
-                augmentation = perturbation_types[augment_name](**augment_kwargs)
+                augmentation_class = perturbation_types[augment_name]
+                if 'global_rank' in augmentation_class.__dict__['__init__'].__code__.co_varnames:
```

Comment:
What in being done here 😨

Reply:
Use inspect module if you really need to test the input argspec

Reply:
Why use the inspect module and add another import when this works and is native Python?

Reply:
Okay, changed to use inspect module